v0/sources/auth0/transform.js

```diff
@@ -12,72 +13,68 @@ const eventNameMap = JSON.parse(
-const isDefinedAndNotNull = x => isDefined(x) && isNotNull(x);
-const removeUndefinedAndNullValues = obj => _.pickBy(obj, isDefinedAndNotNull);
+const prepareIdentifyPayload = event => {
+  const message = new Message("Auth0");
+  message.setEventType("identify");
+  message.setPropertiesV2(event, mapping);
+  return message;
+};
 
-function processEvent(event) {
-  const messageType = "track";
-
-  if (event.type) {
-    const eventType = event.type;
-    const message = new Message(`Auth0`);
-
-    // since only email status events are supported, event type is always track
-    message.setEventType(messageType);
-
-    const eventName = eventNameMap[eventType] || eventType;
-    message.setEventName(eventName);
-
-    message.setProperties(event, mapping);
-
-    if (event.date) {
-      message.setProperty("originalTimestamp", event.date);
-      message.setProperty("sentAt", event.date);
-    }
+const prepareTrackPayload = event => {
+  const message = new Message("Auth0");
+  message.setEventType("track");
+  const eventType = event.type;
+  const eventName = eventNameMap[eventType] || eventType;
+  message.setEventName(eventName);
+  message.setPropertiesV2(event, mapping);
+  return message;
+};
 
-    if (message.context && message.context.traits) {
-      message.userId = message.context.traits.userId;
-    }
+const prepareGroupPayload = event => {
+  const message = new Message("Auth0");
+  message.setEventType("group");
+  message.setPropertiesV2(event, mapping);
+  message.groupId = getGroupId(event);
+  return message;
+};
 
-    if (message.userId && message.userId !== "") {
-      return message;
-    }
+function processEvent(event) {
+  // Dropping the event if type is not present
+  if (!event.type) {
     return null;
   }
-  throw new Error("Unknwon event type from Auth0");
+  const eventType = event.type;
+  // ss -> successful signup
+  if (eventType === "ss") {
+    return prepareIdentifyPayload(event);
+  }
+  // sapi -> Success API Operation
+  if (eventType === "sapi") {
+    if (
+      event.description &&
+      event.description === "Add members to an organization"
+    ) {
+      return prepareGroupPayload(event);
+    }
+    return prepareTrackPayload(event);
+  }
+  return prepareTrackPayload(event);
 }
 
 function process(events) {
   const responses = [];
   let eventList = events;
   if (!Array.isArray(events)) {
-    eventList = [events];
+    eventList = events.logs ? events.logs : [events];
   }
   eventList.forEach(event => {
-    try {
-      const resp = processEvent(event);
-      if (resp) {
-        responses.push(removeUndefinedAndNullValues(resp));
-      }
-    } catch (error) {
-      // TODO: figure out a way to handle partial failures within batch
-      // responses.push({
-      //   statusCode: 400,
-      //   error: error.message || "Unknwon error"
-      // });
+    const resp = processEvent(event.data);
+    if (resp && resp.userId && resp.userId !== "") {
```

Comment:
You can simplify and also a redundant check.
```suggestion
    if (resp?.userId) {
```

Reply:
updated